docs/dev/table/connectors/formats/index.md

```diff
@@ -90,3 +90,62 @@ Flink supports the following formats:
         </tr>
     </tbody>
 </table>
+
+
+Transform Table Format Resources
+--------------------------------
+
+Flink uses Java's [Service Provider Interfaces (SPI)](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) to load the table format factories by their identifiers. Since the SPI resource file named `org.apache.flink.table.factories.Factory` for every table format is under the same directory `META-INF/services`, these resource files will override each other when build the uber-jar of the project which uses more than one table format, which will cause Flink to fail to load table format factories. In this situation, a recommended way is transforming these resource files under the directory `META-INF/services` by [ServicesResourceTransformer](https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html) of maven shade plugin.
```

Comment:
Can we move these document to connector/index?
I think this is common mechanism for all connectors and formats.